### PR TITLE
HSLdevcom/bultti#288 Fix for HFP downloading

### LINKED ISSUE #288: Fix HFP-loader to finish loading data from a date range
Steps:
1) Make any new inspection date range
2) Click Load HFP button
3) After loading, status indicator stays yellow

How to solve: hfpDataStatus should get updated as READY when HFP-loading finishes.

### PR DESCRIPTION
**Pull request by danieldunderfelt**
Fixes #288 

- Move code around until it agreed to download the HFP blobs.

### CODE REVIEW

File: src/hfp/hfpStorage.ts [hunk 1]
```diff
@@ -25,50 +26,74 @@ export async function getHfpBlobs(container, date: string) {
...
-              // Resolve with the path and cleanup function when the file is downloaded.
-              resolve({
-                path: blobPath,
-                cleanup,
+          // Download blobs serially
+          blobDownloadPromise = blobDownloadPromise.then(() =>
+            downloadBlobIntoFile(container, blobName, blobPath)
+              .then(() => {
+                blobPaths.push(blobPath)
+                console.log(`Downloaded blob [${blobIdx}] for ${date}.`)
```
Comment (Divinare) [inline src/hfp/hfpStorage.ts:77]:
```suggestion
                console.log(`Downloaded blob [${dateBlobs.indexOf(blobName) + 1}] for ${date}.`)
```

**Comment by Divinare** [comment]
Testing, failed to load data from: 05 Apr 2021 - 02 May 2021
Time it took to download blobs: 1h
Time it took to insert with 30 concurrent inserts: 45 min (cancelled the process), it should take this long)

Proposal:
Can you make it insert day by date, asyncronously inside date? Currently it spams too much logging and I actually cannot follow where the process is at.

Logging could be something like:
Inserting data from [2021-04-01] blobs...
Inserting data from [2021-04-02] blobs...
...
Inserts completed in ...

And maybe could log how much time has passed after each date blob insertion?

**Comment by Divinare** [comment]
After I shut down worker, it got stuck in inspectionDate.hfpDataStatus = 'LOADING'
Will this ever start loading? If not, we should make the status "NOT_LOADED" somehow if this happens


Tested also with 100ms timeout
```
timeoutRef = setTimeout(() => {
  didTimeOut = true
  blobDownloadPromise = Promise.resolve()
}, 100)
```
Seems to work, the task started and finished quickly, status was NOT_LOADED after which what I expected.

**Comment by danieldunderfelt** [comment]
Reverted to serial date importing. All the logging proposed is already in place, it was just a bit messy.

if you shut down the process then yeah, there is nothing that would set it back to NOT_LOADED.

**Review by Divinare — APPROVED** [review]
Now the DL seemed to work. Loaded HFP-data from 2 week and got 22 GB of data.... The operation took 3500s I think

**Pull request merged**